packages/eventual-send/src/async-contexts/2-sync-context-shallow.js

```diff
@@ -0,0 +1,18 @@
+export const makeSyncContext = () => {
+  let state;
+
+  return harden({
+    run: (val, cb, args = []) => {
+      const prev = state;
+      try {
+        state = val;
+        return cb(...args);
+      } finally {
+        state = prev;
+      }
+    },
+
+    get: () => state,
+  });
+};
+harden(makeSyncContext);
```

Comment:
I'm not sure this is useful, but "thinking out loud" in code.
Here's a wrap function for 2-sync-context-shallow where calling wrap doesn't mutate any globals, but uses message passing to trigger local mutations.
Creating a context appends to a global in the implementation, but this feels meaningfully different so I thought I'd share.
https://gist.github.com/naugtur/72c7a73314a99892b2bbc9fb43d78d1f